nautobot/extras/filters.py

```diff
@@ -823,6 +825,38 @@ def _local_context_data(self, queryset, name, value):
         return queryset.exclude(local_context_data__isnull=value)
 
 
+class NoteFilterSet(BaseFilterSet):
+    q = SearchFilter(
+        filter_predicates={
+            "user_name": "icontains",
+            "note": "icontains",
+            "assigned_object_id": "exact",
+        },
+    )
+    assigned_object_type = ContentTypeFilter()
+    user_id = django_filters.ModelMultipleChoiceFilter(
+        queryset=get_user_model().objects.all(),
+        label="User (ID)",
+    )
+    user = django_filters.ModelMultipleChoiceFilter(
+        field_name="user__username",
+        queryset=get_user_model().objects.all(),
+        to_field_name="username",
+        label="User name",
+    )
```

Comment:
Should be able to combine these two into a single `ForeignKeyOrPKMultipleChoiceFilter`.

Reply:
Okay, just going off of examples. I can change this.

Reply:
Yeah, it's new functionality in 1.4 and we haven't retrofitted all existing filtersets to use this yet - but we want to try to be consistent about it in new code at least. :-)